state-chain/pallets/cf-pools/src/lib.rs

```diff
@@ -986,6 +997,52 @@ pub mod pallet {
 
 			Ok(())
 		}
+
+		/// Cancels a limit order.
+		/// Main difference with `update_limit_order` is that this will not error if the order does
+		/// not exist. This is used for auto-expiring orders that may of already been swept and
+		/// removed, without spamming failure events.
+		#[pallet::call_index(12)]
+		#[pallet::weight(T::WeightInfo::cancel_limit_order())]
+		pub fn cancel_limit_order(
+			origin: OriginFor<T>,
+			base_asset: any::Asset,
+			quote_asset: any::Asset,
+			side: Side,
+			id: OrderId,
+		) -> DispatchResult {
+			let lp = &T::AccountRoleRegistry::ensure_liquidity_provider(origin)?;
+			ensure!(
+				T::SafeMode::get().limit_order_update_enabled,
+				Error::<T>::UpdatingLimitOrdersDisabled
+			);
+
+			let asset_pair = AssetPair::try_new::<T>(base_asset, quote_asset)?;
+			Self::try_mutate_pool(asset_pair, |asset_pair, pool| {
+				match pool.limit_orders_cache[side.to_sold_pair()]
+					.get(lp)
+					.and_then(|limit_orders| limit_orders.get(&id))
+					.cloned()
+				{
+					None => Ok::<(), DispatchError>(()),
+					Some(previous_tick) => {
+						Self::inner_update_limit_order_at_tick(
+							pool,
+							lp,
+							asset_pair,
+							side,
+							id,
+							previous_tick,
+							crate::pallet::IncreaseOrDecrease::Decrease(Amount::MAX),
```

Comment:
Nit, but `IncreaseOrDecrease` and `NoOpStatus` should already be in scope, so you don't need the full path.